src/rgw/rgw_iam_policy.cc

```diff
@@ -1560,6 +1560,15 @@ Effect Policy::eval(const Environment& e,
   return allowed ? Effect::Allow : Effect::Pass;
 }
 
+bool Policy::has_conditional(const string& conditional, bool partial) const {
+  for (const auto&s: statements){
+    for (const auto&c: s.conditions)
+      if (c.has_key(conditional, partial))
```

Comment:
I'd use any_of here, but that's just me.

Reply:
Thinking for the future it might be worth having some condition keys be evaluated lazily, though that might make more sense in some future highly asynchronous RGW where eval can return a future.